File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        ans = []
        num = 0
        if len(tokens) == 1:
            return int(tokens[0])
        for token in tokens:
            if not token in "+-*/":
                ans.append(int(token))
            else:
                string = f"{ans[-2]} {token} {ans[-1]}"
                num = int(eval(string))
                ans.pop()
                ans.pop()
                ans.append(num)
        return num

File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        vector<int> stack;
        for (string token : tokens) {
            if (token == "+" || token == "-" || token == "*" || token == "/") {
                int b = stack.back();
                stack.pop_back();
                int a = stack.back();
                stack.pop_back();
                if (token == "+") {
                    stack.push_back(a + b);
                } else if (token == "-") {
                    stack.push_back(a - b);
                } else if (token == "*") {
                    stack.push_back(a * b);
                } else {
                    stack.push_back(a / b);
                }
            } else {
                stack.push_back(stoi(token));
            }
        }
        return stack[0];
    }
};